vos/perftest/locustfile.py

```diff
@@ -0,0 +1,175 @@
+      else:
+          total_time = int((time.time() - start_time) * 1000)
+          events.request_success.fire(request_type='vcp', name=name, response_time=total_time, response_length=size)
+          # In this example, we've hardcoded response_length=0. If we would want the response length to be 
+          # reported correctly in the statistics, we would probably need to hook in at a lower level
+
+
+def stopwatch(func):
+    """
+    Wrapper to report a function execution time to locust
+    :param func:
+    :return:
+    """
+    def wrapper(*args, **kwargs):
+        # get task's function name
+        previous_frame = inspect.currentframe().f_back
+        _, _, task_name, _, _ = inspect.getframeinfo(previous_frame)
+
+        start = time.time()
+        result = None
+        try:
+            result = func(*args, **kwargs)
+        except Exception as e:
+            total = int((time.time() - start) * 1000)
+            events.request_failure.fire(request_type="TYPE",
+                                        name=func.__name__,
+                                        response_time=total,
+                                        response_length=0,
+                                        exception=e)
+        else:
+            total = int((time.time() - start) * 1000)
+            events.request_success.fire(request_type="TYPE",
+                                        name=func.__name__,
+                                        response_time=total,
+                                        response_length=0)
+        return result
+
+    return wrapper
+
+class Global(User):
+    wait_time = constant_pacing(60/call_freq)
+
+    def __init__(self, *args, **kwargs):
+        super(Global, self).__init__(*args, **kwargs)
+        self.global_tester = GlobalTester()
+
+    @task
+    @stopwatch
+    def anon_get_pub(self):
+        """
+        Negotiate a transfer request for a random public file
+        :return:
+        """
+        self.global_tester.anon_get_pub()
+
+    @task
+    @stopwatch
+    def anon_get_priv(self):
+        """
+        Unauthorized failures
+        :return:
+        """
+        self.global_tester.anon_get_priv()
+
+    @task
+    @stopwatch
+    def auth_get_pub(self):
+        """
+        Negotiate a transfer request for a random public file
+        :return:
+        """
+        self.global_tester.auth_get_pub()
+
+    @task
+    @stopwatch
+    def auth_get_priv(self):
+        self.global_tester.auth_get_priv()
+
+    @task
+    @stopwatch
+    def anon_put(self):
+        """
+        Negotiate a transfer request for pushing a file anon - Failure
+        :return:
+        """
+        self.global_tester.anon_put()
+
+    @task
+    @stopwatch
+    def auth_put(self):
+        """
+        Negotiate a transfer request for pushing a file
+        :return:
+        """
+        self.global_tester.auth_put()
+
+
+# class ApiUser(VCPLocust):
+#   wait_time = between(0.1, 1)
```

Comment:
If you don't need it just remove it.

Reply:
There's a second part to this involving `v*` commands to local storage. Just waiting to see if I need it there.